### PR TITLE
Fix sklearn-dev build

### PR DESCRIPTION
**Pull request by jrbourbeau**
The `r2_score` metric in scikit-learn was recently updated (see https://github.com/scikit-learn/scikit-learn/pull/12855) to raise a warning when passed single sample. This is causing our `sklearn-dev` build to fail. 

This PR updates `test_search_cv_results_none_param` in `tests/model_selection/dask_searchcv/test_model_selection_sklearn.py` to not pass a single sample to `r2_score`.

### CODE REVIEW

**Comment by jrbourbeau** [comment]
Looks like the test failures here are related to the new pandas 0.24 release. I suspect it has to do with `SparseArray`s now implementing the `ExtensionArray` interface. Any thoughts @TomAugspurger?

**Comment by TomAugspurger** [comment]
That look right @jrbourbeau.

I think we may need to update the tests to something like

```python
if sparse:
    dtype = SparseDtype(dtype, dtype(0))
```

just before the dtype assertion.

**Comment by jrbourbeau** [comment]
Thanks @TomAugspurger!
